senior-knights/course-schedulizer
# Connect Harmony with Schedulizer

In this PR:

- creates a global results store for harmony results
- converts the harmony output to our schedulizer format
- displays harmony output on a visual schedule
- displays a button to send the created schedule to the schedulizer
- separates schedule components to separate files
- move helper functions below components in same file
- rename props to be `ComponentNameProps` rather than a shadow of `ComponentName`. This allows for easier imports without name conflicts
- adds `readonly` prop to `Schedule` component that does not do anything yet
- add `useAppContext` hook
- add `useRedirect` hook


To test:
- go to harmony page
- add two professors
- add four courses with the format prefix-number-letter
- add two rooms
- add times with the format xx:xx
- click submit on all of the four forms
- then select all checkboxes for the assignments
- then click `find schedule` and a schedule should be found
- then click `send to schedulizer` and the new schedule should show up on the schedulizer page

Right now, Harmony is quite cumbersome to use, do you have suggestions on how to fix that? Maybe a select all checkboxes button and separating the different parts to their own pages with a `PageStepper` component?

Thanks for the comments and critiques!

closes #136
